plugins/modules/x509_certificate_revocation_info.py

```diff
@@ -0,0 +1,365 @@
+
+        # Check revoced certificates
+        if 'crl_contained' not in result:
+            result['crl_contained'] = False
+            result['crl_record'] = None
+        for cert in crl:
+            if cert.serial_number == self.cert_serial_number:
+                result['crl_contained'] = True
+                result['crl_record'] = cryptography_dump_revoked(cryptography_decode_revoked_certificate(cert))
+                result['crl_source'] = crl_source
+                result['revoked'] = True
+
+    def _report_error(self, soft_fail, msg):
+        if soft_fail:
+            self.module.fail_json(msg=msg)
+        else:
+            self.module.warn(msg)
+
+    def _check_crl_url(self, result, crl_url, soft_fail=False):
+        resp, info = fetch_url(self.module, crl_url, method='GET')
+        if info['status'] != 200:
+            self._report_error(soft_fail, 'HTTP error while loading CRL from {0}: {1}'.format(crl_url, info['status']))
+        else:
+            try:
+                crl_blob = resp.read()
+            except AttributeError as e:
+                self._report_error(soft_fail, 'Error while loading CRL from {0}: {1}'.format(crl_url, to_native(e)))
+                crl_blob = None
+            if crl_blob is not None:
+                self._check_crl(result, crl_blob, crl_url)
+
+    def check_revocation(self):
+        result = dict()
+        result['revoked'] = False
+
+        if self.module.params['crl_path'] is not None:
+            crl_path = self.module.params['crl_path']
+            try:
+                with open(crl_path, 'rb') as f:
+                    crl_blob = f.read()
+            except Exception as e:
+                self.module.fail_json(msg='Error while reading CRL file from disk: {0}'.format(e))
+            self._check_crl(result, crl_blob, crl_source=crl_path)
+
+        if self.module.params['crl_url'] is not None:
+            self._check_crl_url(result, self.module.params['crl_url'])
+
+        if self.module.params['crl_from_cert'] != 'ignore':
+            soft_fail = (self.module.params['crl_from_cert'] == 'check_soft_fail')
+            ext = None
+            try:
+                ext = self.cert.extensions.get_extension_for_class(x509.CRLDistributionPoints)
+            except x509.ExtensionNotFound:
+                pass
+            if ext is None:
+                self._report_error(soft_fail, 'No CRL Distribution Points extension found in certificate')
+            else:
+                for distribution_point in ext.value:
+                    if distribution_point.relative_name is not None:
+                        self._report_error(soft_fail, 'Distribution point with relative name found in certificate')
+                    if distribution_point.full_name is not None:
+                        had_crl_url = False
+                        for name in distribution_point.full_name:
+                            if isinstance(name, x509.UniformResourceIdentifier):
```

Comment:
It's possible for a full name to be an LDAP entry (e.g. would show up in windows cert browser as "URL=ldap://blah", not sure about how it gets decoded by python).

Reply:
I guess it would error out with "`ldap` is an unsupported URL schema", or something like that. Which I think is acceptable.